sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_models.py

```diff
@@ -6,7 +6,141 @@
 
 if TYPE_CHECKING:
     from . import EncryptionAlgorithm, KeyWrapAlgorithm, SignatureAlgorithm
-    from typing import Any
+
+
+class EncryptionAlgorithmConfiguration():
+    """An encryption algorithm, including parameters applicable to using it for encryption.
+    """
+
+    def __init__(self, algorithm, **kwargs):
+        self.algorithm = algorithm
+        self.iv = kwargs.pop("iv", None)
+        self.additional_authenticated_data = kwargs.pop("additional_authenticated_data", None)
```

Comment:
Question here: it's easy to imagine a user using these constructors directly. In that case, should we use the argument validation helper from `_client.py` in `__init__`? We could also validate arguments during calls to `encrypt`/`decrypt` like we do when an `EncryptionAlgorithm` is provided, but that wouldn't give us the chance to raise an error when the configuration is defined.